examples/name-service/programs/name-service/src/lib.rs

```diff
@@ -4,206 +4,42 @@ use anchor_lang::prelude::*;
-        let signer_seeds = [signer_seed, &[bump]];
-
-        let inputs = create_cpi_inputs_for_new_account(
-            proof,
-            new_address_params,
-            compressed_account,
-            &signer_seeds,
-            cpi_context,
-        );
-
-        verify(ctx, &inputs, &[&signer_seeds])?;
+        ctx.light_accounts.record.owner = ctx.accounts.signer.key();
+        ctx.light_accounts.record.name = name;
+        ctx.light_accounts.record.rdata = rdata;
 
         Ok(())
     }
 
-    #[allow(clippy::too_many_arguments)]
     pub fn update_record<'info>(
-        ctx: Context<'_, '_, '_, 'info, NameService<'info>>,
-        proof: CompressedProof,
-        merkle_context: PackedMerkleContext,
-        merkle_tree_root_index: u16,
-        address_merkle_context: PackedAddressMerkleContext,
-        name: String,
-        old_rdata: RData,
+        ctx: LightContext<'_, '_, '_, 'info, UpdateRecord<'info>>,
         new_rdata: RData,
-        cpi_context: Option<CompressedCpiContext>,
     ) -> Result<()> {
-        let unpacked_address_merkle_context =
-            unpack_address_merkle_context(address_merkle_context, ctx.remaining_accounts);
-        let address_seed = derive_address_seed(
-            &[
-                ctx.accounts.signer.key.to_bytes().as_slice(),
-                name.as_bytes(),
-            ],
-            &crate::ID,
-            &unpacked_address_merkle_context,
-        );
-
-        let owner = ctx.accounts.signer.key();
-
-        // Re-create the old compressed account. It's needed as an input for
-        // validation and nullification.
-        let old_record = NameRecord {
-            owner,
-            name: name.clone(),
-            rdata: old_rdata,
-        };
-        let old_compressed_account = input_compressed_account(
-            &old_record,
-            &address_seed,
-            &crate::ID,
-            &merkle_context,
-            merkle_tree_root_index,
-            &address_merkle_context,
-            ctx.remaining_accounts,
-        )?;
-
-        let new_record = NameRecord {
-            owner,
-            name,
-            rdata: new_rdata,
-        };
-        let new_compressed_account = output_compressed_account(
-            &new_record,
-            &address_seed,
-            &crate::ID,
-            &merkle_context,
-            &address_merkle_context,
-            ctx.remaining_accounts,
-        )?;
-
-        let signer_seed = b"cpi_signer".as_slice();
-        let bump = Pubkey::find_program_address(&[signer_seed], &ctx.accounts.self_program.key()).1;
-        let signer_seeds = [signer_seed, &[bump]];
-
-        let inputs = create_cpi_inputs_for_account_update(
-            proof,
-            old_compressed_account,
-            new_compressed_account,
-            &signer_seeds,
-            cpi_context,
-        );
-
-        verify(ctx, &inputs, &[&signer_seeds])?;
```

Comment:
I assume that verify is now always called at the end of the instruction which makes sense.
There is a special case when you want to deal with compressed accounts owned by different programs, ie update a compressed pda and do a compressed token transfer in the same instruction.
This can be achieved by either sending two zkps or using the cpi_context feature.
In either case we need to call verify twice in the instruction.